horreum-backend/src/main/resources/db/changeLog.xml

```diff
@@ -2158,5 +2158,42 @@
             CREATE TRIGGER after_run_update AFTER INSERT OR UPDATE OF data ON run FOR EACH ROW EXECUTE FUNCTION after_run_update_func();
         </sql>
     </changeSet>
+
+    <changeSet id="52" author="jwhiting">
+        <createSequence sequenceName="dataset_id_seq" startValue="1" incrementBy="1" cacheSize="1" />
+        <createTable tableName="dataset">
+            <column name="id" type="integer">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+            <column name="data" type="jsonb">
+                <constraints nullable="false" />
+            </column>
+            <column name="start" type="timestamp without time zone">
+                <constraints nullable="false" />
+            </column>
+            <column name="stop" type="timestamp without time zone">
+                <constraints nullable="false" />
+            </column>
+            <column name="description" type="text" />
+            <column name="testid" type="integer">
+                <constraints nullable="false" />
+            </column>
+            <column name="run_id" type="integer">
+                <constraints nullable="false" />
+            </column>
+            <column name="owner" type="text">
```

Comment:
I don't see owner and access mapped in the entity.

Reply:
Should `DataSet` entity extend from the same `ProtectedBaseEntity` abstract class to inherit the same semantics as `Run`?  @rvansa A quick question about current `Test` and `Run` entities? Is there a reason that `Test` doe not extend `ProtectedBaseEntity`? `Test` can have multiple `TestToken` tokens registered, but `Run` can only have a single String `token`, why are the token security semantics different for `Test` and `Run`?

Reply:
The system of tokens is going to change anyway as it doesn't suit the inteded purpose very well. My current plan is that you could have a token that would grant access to multiple items, so that with one token you could grant access to both Run (or DataSet) and Test.

@whitingjr I think that in the policies you could ignore any tokens for now. That would probably mean not extending `ProtectedBaseEntity` as we would have unused `token` column.

`Test` originally extended `ProtectedBaseEntity` and used single token but I have changed that when converting to multiple tokens. I didn't want to tackle `Run` at the same time.

Reply:
I added an intermediate super-class OwnedEntityBase to hold owner and access. ProtectedEntityBase extends that.